geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/security/ProtobufSimpleAuthenticator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.protocol.protobuf.security;
+
+import static org.apache.geode.internal.protocol.protobuf.ProtocolErrorCode.AUTHENTICATION_FAILED;
+
+import org.apache.geode.internal.protocol.protobuf.AuthenticationAPI;
+
+import org.apache.geode.internal.protocol.protobuf.BasicTypes;
+import org.apache.geode.internal.protocol.protobuf.ClientProtocol;
+import org.apache.geode.internal.security.SecurityService;
+import org.apache.geode.security.AuthenticationRequiredException;
+import org.apache.geode.security.AuthenticationFailedException;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+
+import org.apache.shiro.subject.Subject;
+
+public class ProtobufSimpleAuthenticator implements Authenticator {
```

Comment:
Could we possible also use generics to our advantage here and change the ` implements Authenticator` to `implements Authenticator<Subject>` which would make the `Authenticator` implementation to look like 
```
public interface Authenticator<AuthenticationTokenType> {
  AuthenticationTokenType authenticate(InputStream inputStream, OutputStream outputStream,
      SecurityService securityService) throws IOException, AuthenticationFailedException;}
```

Reply:
Honestly, I hate the fact that we're even exposing this token and see it as a form of broken encapsulation and leaking information about the security implementation.  I don't see a good way around this, but providing more information about the type of the token seems to be going the wrong way.